src/main/scala/com/gu/identity/play/AuthenticatedIdUser.scala

```diff
@@ -82,13 +82,15 @@ object AccessCredentials {
 
       // Adapted from https://github.com/guardian/identity/blob/8663b03/identity-api-client-lib/src/main/java/com/gu/identity/client/IdentityApiClient.java#L321-L334
       def extractUserDataFromToken(tokenString: String) = {
-        val cryptoAccessToken = collectionSigner.getValueForSignedStringJava(tokenString, classOf[CryptoAccessToken])
-
-        if (cryptoAccessToken.expiryTime < DateTime.now) {
-          Left(s"Token: $tokenString has expired")
-        } else if (cryptoAccessToken.targetClient != targetClientId) {
-          Left(s"Token: $tokenString was not targeted for the client '$targetClientId'")
-        } else Right(cryptoAccessToken.getUser)
+        val cryptoAccessToken = collectionSigner.getValueForSignedString[CryptoAccessToken](tokenString)
+
+        cryptoAccessToken map { cryptoToken =>
```

Comment:
It looks like this might change the return type of this function, would you mind setting the return type explicitly on `extractUserDataFromToken()` so I can see what it is?